### PR TITLE
[WIP] wire operator metrics through api monitoring

### PR DESCRIPTION
**Pull request by juanvallejo**
Work in progress...
Adds operator metrics monitoring.
Opening now to begin getting an idea on how to best approach this.

cc @soltysh

### CODE REVIEW

File: pkg/monitor/operator.go [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+				totalRequestCount += metric.Counter.GetValue()
+			}
+
+			qpsLimit := maxQPSAllowed
+			if customLimit, ok := perComponentNamespaceMaxQPSAllowed[pod.Namespace]; ok {
+				qpsLimit = customLimit
+			}
+			qps := totalRequestCount / float64(totalProcTimeSeconds)
+
+			if qps > qpsLimit {
```
Comment (smarterclayton) [inline pkg/monitor/operator.go]: Why is this not an e2e test?

Please explain why your test can't be equivalent to prometheus tests that grab qps metrics and check them, something exactly like `should report telemetry if a cloud.openshift.com token is present`

**Comment by smarterclayton** [comment]
/hold

Monitoring is not the appropriate place for this sort of test.

File: pkg/monitor/operator.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/monitor/operator.go]: @smarterclayton Thanks, opened https://github.com/openshift/origin/pull/21852 to attempt to implement this

**Pull request closed without merging**